presto-pinot/src/main/java/io/prestosql/pinot/PinotBrokerPageSource.java

```diff
@@ -126,6 +128,9 @@ public Page getNextPage()
             blocks[i] = columnBuilders[i].build();
             columnBuilders[i] = columnBuilders[i].newBlockBuilderLike(null);
         }
+        if (decoders.isEmpty()) {
+            return new Page(rowCount);
```

Comment:
Can we add a test for this bug? Also, add an explanation to the commit message describing what was the issue.